# Monte Carlo Simulation for Estimating the Price of a European Put Option

A Monte Carlo simulation is used to price a European put option under risk-neutral measure, simulating N paths of the underlying via Geometric Brownian Motion and taking the discounted expected payoff. Three estimators are compared: a vanilla MC estimator, an antithetic variates estimator, and a control variates estimator, with the Black-Scholes analytical solution serving as the benchmark.

In [1]:
cd -q ../

In [ ]:
from src.pricers.european_options import european_opt_pricer, bs_analytical_solution

In [3]:
S0 = 100
K = 100
r = 0.05
sigma = 0.2
T = 1
N = 1000000

## Analytical Price

In [4]:
bs_analytical_solution(S0, K, r, sigma, T)

10.451

# MC Estimations

### Simple MC estimate for 95% CI:

In [5]:
european_opt_pricer(N, S0, K, r, sigma, T)

(10.453, (10.424, 10.482))

In [6]:
european_opt_pricer(N, S0, K, r, sigma, T, antithetic=True)

(10.457, (10.436, 10.477))

In [7]:
european_opt_pricer(N, S0, K, r, sigma, T, control=True)

(10.451, (10.44, 10.462))

All three MC estimators converge tightly to the Black-Scholes benchmark of 10.451, with point estimates within 0.006. Both variance reduction techniques outperform vanilla (±0.029), with antithetic variates narrowing the interval to ±0.021 and control variates achieving the tightest interval at ±0.011, the latter reflecting the strong positive correlation between the call payoff and the martingale control $e^{-rT}S_T$​ under optimal $\beta^*$. The control variate estimator also exactly matches the Black-Scholes benchmark at 10.451, consistent with the regression correction removing systematic path bias.